addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2733,10 +2733,12 @@ var SnippetsMenu = Widget.extend({
      * @private
      */
     _activateEmptyOptionsTab() {
-        this._updateRightPanelContent({
-            content: this.emptyOptionsTabContent,
-            tab: this.tabs.OPTIONS,
-            forceEmptyTab: true,
+        this._activateSnippet(false).then(() => {
+            this._updateRightPanelContent({
+                content: this.emptyOptionsTabContent,
+                tab: this.tabs.OPTIONS,
+                forceEmptyTab: true,
+            });
```

Comment:
@bvr-odoo A method not returning anything that now returns a promise would be ok, depending on the use case. To avoid of course since people using it would not consider that async part.

However here I don't think you can even call `_activateSnippet` in there since it is currently used like that:

![image](https://user-images.githubusercontent.com/10338094/161580733-d5575a61-d565-4c4f-824b-7b6a38ecb278.png)

-> = call inside the `mutex.exec`... but the `_activateSnippet` call you are adding calls `mutex.exec` too... I guess it works since you don't return the promise here... but it's kinda messy :grimacing: I think that as a stable fix, I would prefer you just don't touch this `_activateEmptyOptionsTab` and let the caller do the `this._activateSnippet(false)` part the right way, if necessary.